From 8490b98b6554d08cc81992b43a0cc9b7c6585cf3 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:30 -0400
Subject: [ethernet] i40e: do not take NVM ownership for SR read

Message-id: <1409658042-22689-210-git-send-email-sassmann@redhat.com>
Patchwork-id: 90396
O-Subject: [RHEL7.1 PATCH 209/281] i40e: do not take NVM ownership for SR read
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Kamil Krawczyk <kamil.krawczyk@intel.com>

We do not need to acquire NVM for Shadow RAM XSUM calculation, as we only
read from SR through SRCTL register for which having the ownership is not
required.

Change-ID: Ie238a8f09917d1d25f24cc7cec271951ac7b98f2
Signed-off-by: Kamil Krawczyk <kamil.krawczyk@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 7a208e83fcedc0b845facc17d05ead0b9b73a967)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_nvm.c b/drivers/net/ethernet/intel/i40e/i40e_nvm.c
index 8129918..66bcb15 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_nvm.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_nvm.c
@@ -324,13 +324,9 @@ i40e_status i40e_validate_nvm_checksum(struct i40e_hw *hw,
  u16 checksum_sr = 0;
  u16 checksum_local = 0;
 
- ret_code = i40e_acquire_nvm(hw, I40E_RESOURCE_READ);
- if (ret_code)
-  goto i40e_validate_nvm_checksum_exit;
-
  ret_code = i40e_calc_nvm_checksum(hw, &checksum_local);
  if (ret_code)
-  goto i40e_validate_nvm_checksum_free;
+  goto i40e_validate_nvm_checksum_exit;
 
  /* Do not use i40e_read_nvm_word() because we do not want to take
   * the synchronization semaphores twice here.
@@ -347,9 +343,6 @@ i40e_status i40e_validate_nvm_checksum(struct i40e_hw *hw,
  if (checksum)
   *checksum = checksum_local;
 
-i40e_validate_nvm_checksum_free:
- i40e_release_nvm(hw);
-
 i40e_validate_nvm_checksum_exit:
  return ret_code;
 }
-- 
1.7.1